scripts/loadEmptyKeys.js

```diff
@@ -0,0 +1,132 @@
+    hideProgress: process.env.HIDE_PROGRESS,
+    dbOptions: {
+        name: process.env.COUCHDB_URL
+    },
+    invokers: {
+        addKeyInDb: {
+            funcName: "gpii.uuidLoader.addKeyInDb",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+        },
+        createNewKeys: {
+            funcName: "gpii.uuidLoader.createNewKeys",
+            args: [
+                "{that}",
+                "{that}.options.totalNumOfKeys",
+                "{that}.options.outputFileName",
+                "{that}.options.hideProgress"
+            ]
+        },
+        dbPut: {
+            funcName: "gpii.uuidLoader.dbPut",
+            args: ["{that}", "{arguments}.0"]
+        }
+    }
+});
+
+gpii.uuidLoader.dbPut = function (that, doc) {
+    var promise = fluid.promise();
+    // We only perform the put after checking that the uuid doesn't exist in DB,
+    // just in case somebody wants to reuse this in the future.
+    //
+    that.get(doc._id).then(function (/* record */) {
+        promise.reject({isError: true, message: "doc with id " + doc._id + " already exists in DB"});
+    }, function (err) {
+        if (err.message === "missing") {
+            that.put(doc).then(function (newDoc) {promise.resolve(newDoc);}, function (err) {promise.reject(err);});
+        } else {
+            promise.reject(err);
+        };
+    });
+    return promise;
+};
+
+gpii.uuidLoader.addKeyInDb = function (that, keyData) {
+    var promise = fluid.promise();
+    var sequence = [
+        that.dbPut(keyData.gpiiKey),
+        that.dbPut(keyData.prefsSafe)
+    ];
+
+    fluid.promise.sequence(sequence).then(function () {
+        promise.resolve(keyData.gpiiKey._id);
+    }, function (error) {
+        promise.reject(error);
+    });
+
+    return promise;
+};
+
+// The real action
+//
+gpii.uuidLoader.createNewKeys = function (that, totalNumOfKeys, outputFileName, hideProgress) {
+    var keyData = gpii.prefsSetsDbUtils.generateKeyData(uuid.v4(), gpii.prefsSetsDbUtils.emptyPreferencesBlock);
+    that.addKeyInDb(keyData).then(function (newKey) {
+        fs.appendFileSync(outputFileName, newKey + "\n");
+        if (!hideProgress) {
+            console.log("Key", gpii.uuidLoader.count, "of", totalNumOfKeys, "created:", newKey);
+        }
+
+        if (gpii.uuidLoader.count < totalNumOfKeys) {
+            gpii.uuidLoader.count++;
+            that.createNewKeys();
+        } else {
+            console.log("We're done adding new keys! :)");
+            console.log("Remember, you have a list of the created keys here:", outputFileName);
+        }
+    }, function (error) {
+        console.log("Got an error, let's stop it here. :/ Error was:", error);
+        console.log("You can find the keys we have created before crashing here:", outputFileName);
+    });
+};
+
+// This triggers the "real action"
+//
+var uuidLoader = gpii.uuidLoader();
+if (uuidLoader.options.totalNumOfKeys > 0) {
```

Comment:
Output a message in the `else` block.

Output an error when required field (`uuidLoader.options.totalNumOfKeys` or `uuidLoader.options.dbOptions.name`) are not provided.

Reply:
The only thing I can do is to check that COUCHDB_URL is empty before instantiating my component, this is because the grade "gpii.pouch" will raise an error if a wrong dbOptions.name is provided. gpii.pouch doesn't provide me a way to catch that error and provide a proper message. Any advice?

Reply:
If so, checking COUCHDB_URL may not be necessary if "gpii.pouch" will raise and error and terminate the process if an empty or incorrect COUCHDB_URL is provided.